server/request.go

```diff
@@ -0,0 +1,70 @@
+package server
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/mysterium/node/identity"
+	"net/http"
+	"net/url"
+)
+
+const (
+	mysteriumAgentName       = "goclient-v0.1"
+	authenticationHeaderName = "Authorization"
+	authenticationSchemaName = "Signature"
+)
+
+var mysteriumApiUrl string
+
+func newGetRequest(path string, params url.Values) (*http.Request, error) {
+	pathWithQuery := fmt.Sprintf("%v?%v", path, params.Encode())
+	return newRequest(http.MethodGet, pathWithQuery, nil)
+}
+
+func newPostRequest(path string, requestBody interface{}) (*http.Request, error) {
+	bodyBytes, err := encodeToJson(requestBody)
+	if err != nil {
+		return nil, err
+	}
+	return newRequest(http.MethodPost, path, bodyBytes)
+}
+
+func newSignedPostRequest(path string, requestBody interface{}, signer identity.Signer) (*http.Request, error) {
+	bodyBytes, err := encodeToJson(requestBody)
```

Comment:
I would just reuse `newPostRequest()` + add header

Reply:
That's the problem - newPostRequest serializes body inside and inserts into request - no clean way to access body bytes later and create signature.